src/ddo/DDO.ts

```diff
@@ -273,4 +303,265 @@ export class DDO {
   public async addSignature(nevermined: Nevermined, publicKey: string) {
     this.proof.signatureValue = await nevermined.utils.signature.signText(this.shortId(), publicKey)
   }
+
+  // DDO Utility functions
+
+  /**
+   * If fins a service condition by name
+   * @param service the service to search in
+   * @param name the name of the condition
+   * @returns the condition
+   */
+  public static findServiceConditionByName(
+    service: Service,
+    name: ConditionType,
+  ): ServiceAgreementTemplateCondition {
+    return service.attributes.serviceAgreementTemplate.conditions.find((c) => c.name === name)
+  }
+
+  /**
+   * Gets the DID in the escrowPayment condition of the service
+   * @param service the service to search in
+   * @returns the DID
+   */
+  public static getDIDFromService(service: Service): string {
+    return `did:nv:${DDO.getParameterFromCondition(service, 'escrowPayment', '_did') as string}`
```

Comment:
this will generate an invalid did if the service does not contain the `escrowPayment` condition